vme/src/db.cpp

```diff
@@ -1368,21 +1302,16 @@ unit_data *read_unit_string(CByteBuffer *pBuf, int type, int len, const char *wh
  */
 void read_unit_file(file_index_type *org_fi, CByteBuffer *pBuf)
 {
-    FILE *f = nullptr;
-    char buf[256];
-
-    snprintf(buf, sizeof(buf), "%s%s.data", g_cServerConfig.getZoneDir().c_str(), org_fi->getZone()->getFilename());
-
-    if ((f = fopen_cache(buf, "rb")) == nullptr)
+    std::filesystem::path buf{g_cServerConfig.getZoneDir()};
+    buf += org_fi->getZone()->getFilename();
+    buf += ".data";
```

Comment:
The filename could be a string zone_type->getDataFileName() or  zone_type->getFileName(".data").

Reply:
I like that idea